File: second-year/second-semester/MPP/mpp-proiect-repository-raulbaciulescu/travelAgency2/model/Utils/Resources.cs


namespace model
{
    public class Resources
    {
        static Resources instance = null;
        private long? selectedId = null;
        public static Resources GetInstance()
        {
            if (instance == null)
                instance = new Resources();
            return instance;
        }
        private Resources() 
        {
        }

        public void SetSelectedId(long flight)
        {
            this.selectedId = flight;
        }

        public long GetSelectedId()
        {
            return (long) this.selectedId;
        }
        
    }
}

File: second-year/second-semester/MPP/mpp-proiect-repository-raulbaciulescu/travelAgency/Client/src/main/java/utils/Resources.java
package utils;

import domain.Flight;
import domain.User;

import java.sql.SQLException;

public class Resources {
    static Resources instance = null;
    private User currentUser;
    private Flight selectedFlight;

    private Resources() {

    }
    public static Resources getInstance() throws SQLException {
        if (instance == null)
            instance = new Resources();
        return instance;
    }

    public void loginCurrentUser(User user) {
        currentUser = user;
    }

    public void logoutCurrentUser() {
        currentUser = null;
    }

    public User getCurrentUser() {
        return currentUser;
    }

    public void setFlight(Flight flight) {
        selectedFlight = flight;
    }

    public Flight getSelectedFlight() {
        return selectedFlight;
    }
}
